main/src/cgeo/geocaching/location/UTMPoint.java

```diff
@@ -0,0 +1,367 @@
+     * values in.
+     */
+    public UTMPoint(final String utmString) {
+        final MatcherWrapper matcher = new MatcherWrapper(PATTERN_UTM, utmString);
+        try {
+            if (matcher.find()) {
+                this.zoneNumber = Integer.parseInt(matcher.group(2));
+                this.zoneLetter = checkZone(matcher.group(3).charAt(0));
+                this.easting = Double.parseDouble(matcher.group(4).replace(",", "."));
+                this.northing = Double.parseDouble(matcher.group(5).replace(",", "."));
+            } else {
+                throw new ParseException("Unable to recognize UTM format in String '" + utmString + "'");
+            }
+        } catch (final NumberFormatException ignored) {
+            throw new ParseException("Error parsing UTM numbers in String '" + utmString + "'");
+        }
+    }
+
+    /**
+     * Constructs a new UTM instance.
+     * 
+     * @param zoneNumber The zone of the coordinate.
+     * @param zoneLetter For UTM, 'A' .. 'Z'
+     * @param easting The easting component.
+     * @param northing The northing component.
+     * @throws NumberFormatException zoneLetter is out of range.
+     */
+    public UTMPoint(final int zoneNumber, final char zoneLetter, final double easting, final double northing) {
+        this.zoneNumber = zoneNumber;
+        this.zoneLetter = checkZone(zoneLetter);
+        this.easting = easting;
+        this.northing = northing;
+    }
+
+    public boolean equals(final Object obj) {
+        if (obj instanceof UTMPoint) {
+            final UTMPoint pnt = (UTMPoint) obj;
+            return northing == pnt.northing
+                    && easting == pnt.easting
+                    && zoneNumber == pnt.zoneNumber
+                    && zoneLetter == pnt.zoneLetter;
+        }
+        return false;
+    }
+
+    /**
+     * Method that provides a check for UTM zone letters. Returns an uppercase
+     * version of any valid letter passed in, 'A' .. 'Z'.
+     * 
+     * @throws NumberFormatException if zone letter is invalid.
+     */
+    private char checkZone(final char inZone) {
+        final char zone = Character.toUpperCase(inZone);
+
+        if (zone < 'A' || zone > 'Z') {
+            throw new NumberFormatException("Invalid UTMPoint zone letter: " + zone);
+        }
+
+        return zone;
+    }
+
+    /**
+     * Returns a string representation of the object.
+     * 
+     * @return String representation
+     */
+    public String toString() {
+        return String.format("%d%c E %d N %d", zoneNumber, zoneLetter, Math.round(easting), Math.round(northing));
+    }
+
+    /**
+     * Converts UTM coords to lat/long.
+     * <p>
+     * Equations from USGS Bulletin 1532 <br>
+     * East Longitudes are positive, West longitudes are negative. <br>
+     * North latitudes are positive, South latitudes are negative.
+     *
+     * @return Geopoint containing the lat/long value, or
+     *         <code>null</code> if conversion failed.
```

Comment:
We support from `null` so much… Wouldn't it make more sense to throw an exception there, and have this function be annotated with `NonNull`?